models/step_current_generator.cpp

```diff
@@ -100,48 +108,126 @@ nest::step_current_generator::Buffers_::Buffers_( const Buffers_&,
 void
 nest::step_current_generator::Parameters_::get( DictionaryDatum& d ) const
 {
-  ( *d )[ names::amplitude_times ] =
-    DoubleVectorDatum( new std::vector< double >( amp_times_ ) );
+  std::vector< double >* times_ms = new std::vector< double >();
+  times_ms->reserve( amp_time_stamps_.size() );
+  for ( std::vector< Time >::const_iterator it = amp_time_stamps_.begin();
+        it != amp_time_stamps_.end();
+        ++it )
+  {
+    times_ms->push_back( it->get_ms() );
+  }
+  ( *d )[ names::amplitude_times ] = DoubleVectorDatum( times_ms );
   ( *d )[ names::amplitude_values ] =
     DoubleVectorDatum( new std::vector< double >( amp_values_ ) );
+  ( *d )[ names::allow_offgrid_times ] = BoolDatum( allow_offgrid_amp_times_ );
+}
+
+nest::Time
+nest::step_current_generator::Parameters_::validate_time_( double t,
+  const Time& t_previous )
+{
+  if ( t == 0.0 )
+  {
+    throw BadProperty( "Amplitude cannot be changed at 0." );
```

Comment:
Preferably update to "Amplitude cannot be changed at t = 0."

Reply:
Fixed.